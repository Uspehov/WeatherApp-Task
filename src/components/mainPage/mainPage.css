.mainPage-container {
    width: 100%;
}

.mainPage-buttonsContainer {
    width: 70%;
    height: 60px;
    display: inline-flex;
    justify-content: space-between;
    word-break: break-all;
}

.mainPage-textContainer {
    margin-bottom: 50px;
}

.mainButton {
    width: 30%;
    background-color: cadetblue;
    color: whitesmoke;
    border: none;
    text-decoration: none;
    display: flex;
    align-items: center;
    justify-content: center;
    cursor: pointer;
}

@media (max-width: 750px) {
    .mainPage-buttonsContainer {height: 100px;}
}