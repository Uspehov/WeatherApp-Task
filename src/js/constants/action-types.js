export const WEATHER = {
    GET_WEATHER: "GET_WEATHER",
    GET_WEATHER_SUCCES: "GET_WEATHER_SUCCES",
    GET_WEATHER_FAILURE: "GET_WEATHER_FAILURE"
};