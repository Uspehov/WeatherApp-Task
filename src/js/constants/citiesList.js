export const citiesArr = ['Moscow', 'Saint Petersburg', 'Rostov-on-Don'];