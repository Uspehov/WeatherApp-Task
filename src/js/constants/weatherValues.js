export const weatherValues = [ '°F', 'Pa', '%', '°F', '°F'];