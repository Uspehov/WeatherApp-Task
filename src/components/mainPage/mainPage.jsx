import React from 'react';
import { Link } from "react-router-dom";
import { citiesArr } from '../../js/constants/citiesList';
import './mainPage.css';

export class MainPage extends React.Component {

    render() {
        return (
            <div className='mainPage-container'>
                <div className='mainPage-textContainer'>
                    <h1>Welcome</h1>
                    <span>Here you can check actual weather for following cities.</span>
                </div>
                <div className='mainPage-buttonsContainer'>
                    {citiesArr.map(item => (<Link key={item} to={`/${item}`} className='mainButton' >{item}</Link>))}
                </div>
            </div>
        );
    }
}
