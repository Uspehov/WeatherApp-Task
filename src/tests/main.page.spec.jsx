import React from 'react';
import  { mount, configure }  from 'enzyme';
import  { MainPage }  from '../components/mainPage/mainPage.jsx';
import { citiesArr } from '../js/constants/citiesList';
import { BrowserRouter as Router } from 'react-router-dom';
import { Link } from "react-router-dom";
import Adapter from 'enzyme-adapter-react-16';

configure({adapter: new Adapter()});

describe('Main Page', () => {

    it('should render', () => {
        const wrapper = mount(<Router><MainPage /></Router>);
        expect(wrapper.debug()).toMatchSnapshot();
    });

    it('it should render proper number of buttons', () => {
        const wrapper = mount(<Router><MainPage /></Router>);
        expect(wrapper.find(Link).length).toBe(citiesArr.length);
    });
});