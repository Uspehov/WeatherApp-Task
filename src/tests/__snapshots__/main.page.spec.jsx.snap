// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Page should render 1`] = `
"<BrowserRouter>
  <Router history={{...}}>
    <MainPage>
      <div className=\\"mainPage-container\\">
        <div className=\\"mainPage-textContainer\\">
          <h1>
            Welcome
          </h1>
          <span>
            Here you can check actual weather for following cities.
          </span>
        </div>
        <div className=\\"mainPage-buttonsContainer\\">
          <Link to=\\"/Moscow\\" className=\\"mainButton\\">
            <LinkAnchor className=\\"mainButton\\" href=\\"/Moscow\\" navigate={[Function: navigate]}>
              <a className=\\"mainButton\\" href=\\"/Moscow\\" onClick={[Function: onClick]}>
                Moscow
              </a>
            </LinkAnchor>
          </Link>
          <Link to=\\"/Saint Petersburg\\" className=\\"mainButton\\">
            <LinkAnchor className=\\"mainButton\\" href=\\"/Saint Petersburg\\" navigate={[Function: navigate]}>
              <a className=\\"mainButton\\" href=\\"/Saint Petersburg\\" onClick={[Function: onClick]}>
                Saint Petersburg
              </a>
            </LinkAnchor>
          </Link>
          <Link to=\\"/Rostov-on-Don\\" className=\\"mainButton\\">
            <LinkAnchor className=\\"mainButton\\" href=\\"/Rostov-on-Don\\" navigate={[Function: navigate]}>
              <a className=\\"mainButton\\" href=\\"/Rostov-on-Don\\" onClick={[Function: onClick]}>
                Rostov-on-Don
              </a>
            </LinkAnchor>
          </Link>
        </div>
      </div>
    </MainPage>
  </Router>
</BrowserRouter>"
`;
